app/(root)/profile/[id]/edit/page.tsx

```diff
@@ -1,7 +1,32 @@
+import EditProfile from "@/components/shared/profile/edit/EditProfile";
+import UploadPhoto from "@/components/shared/profile/edit/UploadPhoto";
+
+import { getOneUser } from "@/lib/actions/user.actions";
+import { getSession } from "@/lib/authOptions";
+import { redirect } from "next/navigation";
 import React from "react";
 
-const EditProfilePage = () => {
-  return <div>Page</div>;
+const EditProfilePage = async () => {
+  const session = await getSession();
+  let user;
+  if (session) {
+    user = await getOneUser(session?.user?.email!);
+    console.log(user);
+  } else {
+    redirect("/sign-in");
+  }
+  const cleanUser = JSON.parse(JSON.stringify(user));
```

Comment:
```suggestion
const EditProfilePage = async () => {
  const session = await getSession();
  if(!session){
    redirect("/sign-in");
    return;
  }
   const user = await getOneUser(session?.user?.email!);
   console.log(user);
  const cleanUser = JSON.parse(JSON.stringify(user));
```

Instead of an if/else, if one of the paths terminates (the redirect), then just check that condition and the rest of your code can simplify.

We also will likely be using the middleware for this, right?